### PR TITLE
Fix docstring of `optuna.integration.lightgbm.train`

### LINKED ISSUE #904: The doc of `optuna.integration.lightgbm.train` function cannot be generated if LightGBM isn't installed.
Due to the bug the title describes, [the current API reference doc](https://optuna.readthedocs.io/en/latest/) doesn't contain the entry of `optuna.integration.lightgbm.train` function.

# Expected behavior

The build of the function doc should be succeeded even if LightGBM isn't installed in the build environment.

## Error messages, stack traces, or logs

```console
$ cd optuna/docs
$ make html
...
WARNING: autodoc: failed to import function 'train' from module 'optuna.integration.lightgbm'; the following exception was raised:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/optuna/envs/latest/lib/python3.7/site-packages/sphinx/util/inspect.py", line 225, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: module 'optuna.integration.lightgbm' has no attribute 'train'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/optuna/envs/latest/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 193, in import_object
    obj = attrgetter(obj, attrname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/optuna/envs/latest/lib/python3.7/site-packages/sphinx/ext/autodoc/__init__.py", line 290, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/optuna/envs/latest/lib/python3.7/site-packages/sphinx/ext/autodoc/__init__.py", line 1563, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/optuna/envs/latest/lib/python3.7/site-packages/sphinx/util/inspect.py", line 241, in safe_getattr
    raise AttributeError(name)
AttributeError: train
...
```

## Steps to reproduce

1. Ensure that LightGBM isn't installed in the built environment.
2. `$ git clone git://github.com/optuna/optuna.git`
3. `$ cd optuna/docs`
4. `$ make html`

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
To fix #904 . (Sorry I closed previous PR #923 ).

### CODE REVIEW

File: optuna/integration/lightgbm.py [hunk 1]
```diff
@@ -28,10 +27,12 @@
             continue
         setattr(sys.modules[__name__], api_name, lgb.__dict__[api_name])
 
+    # API from lightgbm_tuner.
     for api_name in _names_from_tuners:
         setattr(sys.modules[__name__], api_name, tuner.__dict__[api_name])
 else:
-    LightGBMTuner = object  # type: ignore
+    # To create docstring of train
```
Comment (toshihikoyanase) [inline optuna/integration/lightgbm.py]: Please add a period at the end of the sentence (c.f., https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences).
```suggestion
    # To create docstring of train.
```

File: optuna/integration/lightgbm_tuner/__init__.py [hunk 2]
```diff
@@ -1,8 +1,15 @@
 from optuna._experimental import experimental
-from optuna.integration.lightgbm_tuner.sklearn import LGBMClassifier, LGBMModel, LGBMRegressor  # NOQA
-from optuna.integration.lightgbm_tuner.optimize import LightGBMTuner
 from optuna import type_checking
 
+try:
+    from optuna.integration.lightgbm_tuner.sklearn import LGBMClassifier, LGBMModel, LGBMRegressor  # NOQA
+    from optuna.integration.lightgbm_tuner.optimize import LightGBMTuner
```
Comment (toshihikoyanase) [inline optuna/integration/lightgbm_tuner/__init__.py]: `_available = True` seems to be required here to pass `_check_lightgbm_availability()`.

Comment (keisuke-umezawa) [reply optuna/integration/lightgbm_tuner/__init__.py]: 🙇

**Comment by smly** [comment]
LGTM:+1:

**Pull request merged**